cli/azd/pkg/azapi/stack_deployments.go

```diff
@@ -0,0 +1,586 @@
+
+	stack := armdeploymentstacks.DeploymentStack{
+		Tags: tags,
+		Properties: &armdeploymentstacks.DeploymentStackProperties{
+			ActionOnUnmanage: &armdeploymentstacks.ActionOnUnmanage{
+				Resources:        &deleteBehavior,
+				ManagementGroups: &deleteBehavior,
+				ResourceGroups:   &deleteBehavior,
+			},
+			DenySettings: &armdeploymentstacks.DenySettings{
+				Mode: convert.RefOf(armdeploymentstacks.DenySettingsModeNone),
+			},
+			Parameters: stackParams,
+			Template:   armTemplate,
+		},
+	}
+	poller, err := client.BeginCreateOrUpdateAtResourceGroup(ctx, resourceGroup, deploymentName, stack, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	_, err = poller.PollUntilDone(ctx, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	return d.GetResourceGroupDeployment(ctx, subscriptionId, resourceGroup, deploymentName)
+}
+
+func (d *deploymentStacks) ListSubscriptionDeploymentOperations(
+	ctx context.Context,
+	subscriptionId string,
+	deploymentName string,
+) ([]*armresources.DeploymentOperation, error) {
+	if err := d.init(ctx); err != nil {
+		return nil, err
+	}
+
+	deployment, err := d.GetSubscriptionDeployment(ctx, subscriptionId, deploymentName)
+	if err != nil {
+		return nil, err
+	}
+
+	deploymentName = filepath.Base(deployment.DeploymentId)
+
+	return d.standardDeployments.ListSubscriptionDeploymentOperations(ctx, subscriptionId, deploymentName)
+}
+
+func (d *deploymentStacks) ListResourceGroupDeploymentOperations(
+	ctx context.Context,
+	subscriptionId string,
+	resourceGroupName string,
+	deploymentName string,
+) ([]*armresources.DeploymentOperation, error) {
+	if err := d.init(ctx); err != nil {
+		return nil, err
+	}
+
+	// The requested deployment name may be an inner deployment which will not be found in the deployment stacks.
+	// If this is the case continue on checking if there is a stack deployment
+	// If a deployment stack is found then use the deployment id of the stack
+	deployment, err := d.GetResourceGroupDeployment(ctx, subscriptionId, resourceGroupName, deploymentName)
+	if !errors.Is(err, ErrDeploymentNotFound) {
+		return nil, err
+	}
+
+	if deployment != nil && deployment.DeploymentId != "" {
+		deploymentName = filepath.Base(deployment.DeploymentId)
+	}
+
+	return d.standardDeployments.ListResourceGroupDeploymentOperations(
+		ctx,
+		subscriptionId,
+		resourceGroupName,
+		deploymentName,
+	)
+}
+
+func (d *deploymentStacks) WhatIfDeployToSubscription(
+	ctx context.Context,
+	subscriptionId string,
+	location string,
+	deploymentName string,
+	armTemplate azure.RawArmTemplate,
+	parameters azure.ArmParameters,
+) (*armresources.WhatIfOperationResult, error) {
+	if err := d.init(ctx); err != nil {
+		return nil, err
+	}
+
+	return d.standardDeployments.WhatIfDeployToSubscription(
+		ctx,
+		subscriptionId,
+		location,
+		deploymentName,
+		armTemplate,
+		parameters,
+	)
+}
+
+func (d *deploymentStacks) WhatIfDeployToResourceGroup(
```

Comment:
I updated deployment stacks to return an `ErrPreviewNotSupported` error.  Keeping the template hash in here for the time being since it is used internally by the deployments.  The StandardDeployments is also really just a wrapper around the ARM deployment API so calling internally back into this from StackDeployments seems reasonable enough right now.